robot-agent/utils.js

```diff
@@ -152,14 +147,26 @@ const startPackage = (name) => {
             })
         });
       subprocess.unref();
-
-      // Start resource monitoring for the package
-      ResourceMonitor.startMonitoring(name, subprocess.pid);
-
-      // start watching status.json (from startPackage) and report in mqtt
-      watchStatus(name, 'requested');
+      packagePid = subprocess.pid;
+    }
+    if(!packagePid) {
+      // Get the PID asynchronously
+      packagePid = await new Promise((resolve) => {
+        pgrep.stdout.on('data', (data) => {         
+          resolve(parseInt(data.toString().trim()));
+        });
```

Comment:
Any chance this was proposed by co-pilot? I'm asking because I think it looks right on the surface but I don't think will reliably work in practice. What makes us believe that stdout will still emit `data` events *after* the process has already exited? I think you need to "collect" the stdout above as it happens and store the received PID in a variable just in case you later need it.

Reply:
Indeed! From https://nodejs.org/api/child_process.html#event-exit:
> When the 'exit' event is triggered, child process stdio streams might still be open.

So the exit event and the output on stdout are completely async from each other. We should probably design the logic accordingly, i.e., wait until both the exit has happened *and* the stdout ReadableStream has closed. Using a sync or promise version of spawn/exec might be simpler here if we can use that safely -- we don't want to block the agent if the command fails or hangs.